test/base/test_history.py

```diff
@@ -620,3 +620,30 @@ def test_hdf5_history_mp():
                             np.testing.assert_array_equal(
                                 mem_entry_trace[iteration],
                                 hdf5_entry_trace[iteration])
+
+
+def test_trim_history():
+    """
+    Test whether the history gets correctly trimmed to be monotonically
+    decreasing.
+    """
+    problem = CRProblem()
+    pypesto_problem = problem.get_problem()
+
+    optimizer = pypesto.optimize.ScipyOptimizer()
+    history_options = pypesto.HistoryOptions(trace_record=True)
+    result = pypesto.optimize.minimize(
+        problem=pypesto_problem, optimizer=optimizer,
+        n_starts=2, history_options=history_options,
+        engine=MultiProcessEngine()
```

Comment:
Why launch two optimizations if only one is used later on?

Reply:
you have a point there, I had a reason but I cannot remember what it was :D